File: app/src/main/java/wannabit/io/cosmostaion/widget/txDetail/TxStakingRewardHolder.java
package wannabit.io.cosmostaion.widget.txDetail;

import android.content.Context;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;

import java.util.ArrayList;

import cosmos.tx.v1beta1.ServiceOuterClass;
import wannabit.io.cosmostaion.R;
import wannabit.io.cosmostaion.base.BaseData;
import wannabit.io.cosmostaion.base.chains.ChainConfig;
import wannabit.io.cosmostaion.model.type.Coin;
import wannabit.io.cosmostaion.utils.WDp;

public class TxStakingRewardHolder extends TxHolder {
    ImageView itemRewardImg;
    TextView itemRewardTitle;
    TextView itemDelegator, itemValidator, itemMoniker, itemRewardAmount, itemRewardAmountDenom;

    public TxStakingRewardHolder(@NonNull View itemView) {
        super(itemView);
        itemRewardImg = itemView.findViewById(R.id.tx_reward_icon);
        itemRewardTitle = itemView.findViewById(R.id.tx_reward_text);
        itemDelegator = itemView.findViewById(R.id.tx_reward_delegator);
        itemValidator = itemView.findViewById(R.id.tx_reward_validator);
        itemMoniker = itemView.findViewById(R.id.tx_reward_moniker);
        itemRewardAmount = itemView.findViewById(R.id.tx_reward_amount);
        itemRewardAmountDenom = itemView.findViewById(R.id.tx_reward_amount_symbol);
    }

    public void onBindMsg(Context c, BaseData baseData, ChainConfig chainConfig, ServiceOuterClass.GetTxResponse response, int position, String address) {
        itemRewardImg.setColorFilter(ContextCompat.getColor(c, chainConfig.chainColor()), android.graphics.PorterDuff.Mode.SRC_IN);

        try {
            cosmos.distribution.v1beta1.Tx.MsgWithdrawDelegatorReward msg = cosmos.distribution.v1beta1.Tx.MsgWithdrawDelegatorReward.parseFrom(response.getTx().getBody().getMessages(position).getValue());
            itemDelegator.setText(msg.getDelegatorAddress());
            itemValidator.setText(msg.getValidatorAddress());
            itemMoniker.setText( "(" + baseData.getValidatorInfo(msg.getValidatorAddress()).getDescription().getMoniker() + ")");

            ArrayList<Coin> rewardsCoins = WDp.onParseStakeReward(response, position);
            WDp.setDpCoin(c, baseData, chainConfig, rewardsCoins.get(0), itemRewardAmountDenom, itemRewardAmount);

        } catch (Exception e) {}

    }
}

File: app/src/main/java/wannabit/io/cosmostaion/widget/txDetail/TxStakingRewardHolder.kt
package wannabit.io.cosmostaion.widget.txDetail

import android.content.Context
import android.graphics.PorterDuff
import android.view.View
import androidx.core.content.ContextCompat
import cosmos.distribution.v1beta1.Tx.MsgWithdrawDelegatorReward
import cosmos.tx.v1beta1.ServiceOuterClass
import wannabit.io.cosmostaion.base.BaseData
import wannabit.io.cosmostaion.base.chains.ChainConfig
import wannabit.io.cosmostaion.databinding.ItemTxRewardBinding
import wannabit.io.cosmostaion.utils.WDp

class TxStakingRewardHolder(itemView: View) : TxHolder(itemView) {

    private lateinit var binding: ItemTxRewardBinding

    override fun onBindMsg(c: Context?, baseData: BaseData?, chainConfig: ChainConfig?, response: ServiceOuterClass.GetTxResponse?, position: Int, address: String?) {
        super.onBindMsg(c, baseData, chainConfig, response, position, address)
        binding = ItemTxRewardBinding.bind(itemView)
        binding.txRewardIcon.setColorFilter(ContextCompat.getColor(c!!, chainConfig!!.chainColor()), PorterDuff.Mode.SRC_IN)

        response?.apply {
            val msg = MsgWithdrawDelegatorReward.parseFrom(response.tx.body.getMessages(position).value)
            binding.txRewardDelegator.text = msg.delegatorAddress
            binding.txRewardValidator.text = msg.validatorAddress
            baseData?.apply {
                binding.txRewardMoniker.text = "(" + baseData.getValidatorInfo(msg.validatorAddress).description.moniker + ")"

                val rewardsCoins = WDp.onParseStakeReward(response, position)
                if (rewardsCoins.size > 0) {
                    binding.rewardLayout0.visibility = View.VISIBLE
                    WDp.setDpCoin(c, baseData, chainConfig, rewardsCoins[0], binding.txRewardSymbol0, binding.txRewardAmount0)
                }

                if (rewardsCoins.size > 1) {
                    binding.rewardLayout1.visibility = View.VISIBLE
                    WDp.setDpCoin(c, baseData, chainConfig, rewardsCoins[1], binding.txRewardSymbol1, binding.txRewardAmount1)
                }

                if (rewardsCoins.size > 2) {
                    binding.rewardLayout2.visibility = View.VISIBLE
                    WDp.setDpCoin(c, baseData, chainConfig, rewardsCoins[2], binding.txRewardSymbol2, binding.txRewardAmount2)
                }

                if (rewardsCoins.size > 3) {
                    binding.rewardLayout3.visibility = View.VISIBLE
                    WDp.setDpCoin(c, baseData, chainConfig, rewardsCoins[3], binding.txRewardSymbol3, binding.txRewardAmount3)
                }
            }
        }
    }
}